src/include/transaction/transaction_manager.h

```diff
@@ -109,5 +106,12 @@ class TransactionManager {
   void DeallocateInsertedTupleIfVarlen(TransactionContext *txn, storage::UndoRecord *undo,
                                        const storage::TupleAccessStrategy &accessor) const;
   void GCLastUpdateOnAbort(TransactionContext *txn);
+
+  /**
+   * Notifies the txn manager that the txn has been serialized and can safely be GC'd. Should only be called by
+   * LogManager. If logging is disabled, the callback will be called by LogCommit/LogAbort
+   * @param txn transaction that has been serialized
+   */
+  void NotifyTransactionSerialized(TransactionContext *txn);
```

Comment:
I don't think this function is the right solution to the race we discussed.  Specifically, we can continue to pass off completed transactions to the GC without causing any issues since we're not releasing the active timestamps until its been serialized.  Additionally, we the log serializer should have a reference to the timestamp manager and make the call to `RemoveTransaction` directly so that the dependency is explicit.

Reply:
Done, changed to remove txn from `completed_txns_` at commit/abort time, and serializer calls to timestamp manager to remove txn from active txn.